citro3d/src/math/matrix.rs

```diff
@@ -215,7 +182,33 @@ impl Matrix4 {
                 camera_up.0,
                 coordinates.is_left_handed(),
             );
-            Self::new(out.assume_init())
+            Self::from_raw(out.assume_init())
         }
     }
 }
+
+impl core::fmt::Debug for Matrix4 {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_tuple("Matrix4").field(&self.rows_wzyx()).finish()
+    }
+}
+impl PartialEq<Matrix4> for Matrix4 {
+    fn eq(&self, other: &Matrix4) -> bool {
+        self.rows_wzyx() == other.rows_wzyx()
+    }
+}
+impl Eq for Matrix4 {}
+
+#[cfg(feature = "glam")]
+impl From<glam::Mat4> for Matrix4 {
+    fn from(mat: glam::Mat4) -> Self {
+        Matrix4::from_rows(core::array::from_fn(|i| mat.row(i).into()))
```

Comment:
because we don't need to pay the cost of the transpose I think was my reasoning? not sure if it actually is a "cost" though compared to this since its all done "manually" in actuality

Reply:
Yeah I guess without digging into the code / looking at generated assembly, it's hard to know which is more optimal. Not a big deal either way, I think it's more important to get functionality in place and optimize later since this library is still pretty early in development.